python/python/lance/indices.py

```diff
@@ -185,6 +197,60 @@ def train_ivf(
             )
             return IvfModel(centroids_array, distance_type)
 
+    def assign_ivf_partitions(
+        self,
+        ivf_model: IvfModel,
+        dst_dataset_uri: Optional[str],
+        accelerator: Union[str, "torch.Device"],
+    ) -> str:
+        """
+        Calculates which IVF partition each vector belongs to.  This searches the
+        IVF centroids and assigns the closest centroid to the vector.  The result is
+        stored in a Lance dataset located at dst_dataset_uri.  The schema of the
+        partition assignment dataset is:
+
+        row_id: uint64
+        partition: uint32
+
+        Note: There is no advantage to separately computing the partition assignment
```

Comment:
Maybe it is also useful for checkpoint purpose.

Reply:
Hmm...perhaps.  Though I think a better way to checkpoint the "transform vectors" stage will be to break up the work by fragments.